File: src/sort/selection-sort.ts
import { Sortable } from "@/model/Sortable";
import { swap } from "./util";


export class SelectionSort implements Sortable {

    sort(arr: number[], maxNum?: number) {
        for (let i = 0; i < arr.length; i++) {
            for (let j = i + 1; j < arr.length; j++) {
                if (arr[i] > arr[j]) {
                    swap(arr, i, j);
                }
            }
        }
        return arr;
    }

}

File: src/main/java/com/github/danimaniarqsoft/sorting/SelectionSort.java
package com.github.danimaniarqsoft.sorting;

public class SelectionSort implements Sort {

    @Override
    public void sort(int[] data) {
        if (data == null || data.length == 1) {
            return;
        }

        for (int i = 0; i < data.length; i++) {
            for (int j = i + 1; j < data.length; j++) {
                if (data[j] < data[i]) {
                    swap(data, j, i);
                }
            }
        }
    }

    private static void swap(int[] arr, int left, int right) {
        int temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
    }

}
